utils/downloads.go

```diff
@@ -0,0 +1,91 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+var snapshots = map[string]string{
+	"mainnet": "https://juno-snapshot.s3.us-east-2.amazonaws.com/mainnet/juno_mainnet_v0.5.0_136902.tar",
+	"goerli":  "https://juno-snapshot.s3.us-east-2.amazonaws.com/goerli/juno_goerli_v0.5.0_839969.tar",
+	"goerli2": "https://juno-snapshot.s3.us-east-2.amazonaws.com/goerli2/juno_goerli2_v0.5.0_135973.tar",
+}
+
+const (
+	downloadBufferSize = 8192 // 8KB
+	dirPermissionMod   = 0o770
+)
+
+type Downloader interface {
+	DownloadAndWrite(ctx context.Context, reader io.Reader, writer io.Writer, bar *pb.ProgressBar) error
+}
+
+type RealDownloader struct{}
+
+func (d *RealDownloader) DownloadAndWrite(ctx context.Context, reader io.Reader, writer io.Writer, bar *pb.ProgressBar) error {
```

Comment:
Why do we need `bar`?

Reply:
For printing the download progress.

Reply:
I see `bar.Finish()`, do we use it elsewhere?